app_test.go

```diff
@@ -2265,3 +2265,239 @@ func TestWhenExitSubCommandWithCodeThenAppQuitUnexpectedly(t *testing.T) {
 		t.Errorf("exitCodeFromOsExiter valeu should be %v, but its value is %v", testCode, exitCodeFromExitErrHandler)
 	}
 }
+
+func TestDuplicateCommandName(t *testing.T) {
+	duplicateCommandName := "hello"
+	os.Args = []string{"say", duplicateCommandName}
+	app := NewApp()
+	app.Name = "say"
+	app.Commands = []Command{
+		{
+			Name: duplicateCommandName,
+			Action: func(c *Context) error {
+				fmt.Println("Hello")
+				return nil
+			},
+		},
+		{
+			Name: duplicateCommandName,
+			Action: func(c *Context) error {
+				fmt.Println("Hello duplicate")
+				return nil
+			},
+		},
+	}
+
+	err := app.Run(os.Args)
```

Comment:
⭐⭐⭐ _(must change)_ echoing my command here https://github.com/urfave/cli/pull/984#discussion_r356391301 the unit you are most interested in testing isn't `app.Run`, it's `checkDuplicateCommandNames`. In service of that, I need to see the majority of these tests running their assertions against `checkDuplicateCommandNames`.